pkg/routeagent_driver/handlers/kubeproxy/endpoint_handler.go

```diff
@@ -138,7 +151,10 @@ func (kp *SyncHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
 		return err
 	}
 
-	kp.updateRoutingRulesForHostNetworkSupport(endpoint.Spec.Subnets, Delete)
+	if kp.localCableDriver != vxlan.CableDriverName {
+		kp.updateRoutingRulesForHostNetworkSupport(endpoint.Spec.Subnets, Delete)
+	}
```

Comment:
Again what happens with multiple cable drivers....  also would be nice if this code does not need knowledge of a specific driver. Perhaps this could be part of the cable driver impl?

Reply:
This is the part of the route-agent, probably we need to add a cable-driver interface there too.

Reply:
yeah we have the `cleanup.Handler` interface. In this case we need `RemoteEndpointCreated`, `RemoteEndpointRemoved` etc - it seems we could extend the `event.Handler` interface to the cable drivers. In fact `cleanup.Handler` is really just a subset of `event.Handler`. Both `GatewayToNonGatewayTransition` and `NonGatewayCleanup` are called from `TransitionToNonGateway`. I propose we eliminate `cleanup.Handler` and use `event.Handler` for the cable driver interactions. The cable driver handlers can be added to the event `Registry` same as the `SyncHandler`.

Reply:
Actually let's back up a bit. Why isn't `updateRoutingRulesForHostNetworkSupport` done for the vxlan driver? Is it b/c it isn't needed but doesn't hurt anything if it is done or it's problematic for the `vxlan` driver? If the latter then how will we handle this when both the `vxlan` and `libreswan` drivers are present?

Reply:
Yes it is done since the route inserted by the vxlan driver ( which can handle hostnetworks too) will conflict with the one added by updateRoutingRulesForHostNetworkSupport. We will only have either vxlan or libreswan between a pair of endpoints so either one of the rule should be present.

Reply:
Will it be possible to remove any hostNetworkSupport specific routes from vxlan driver and let them be handled in this API, just like how it works for other drivers (libreswan, wireguard etc)?